TODO

```diff
@@ -0,0 +1,29 @@
+- Set permissions on the named pipe/Unix socket correctly. No need to let
+  random users spy on worktrees' activity to which they have no read access.
+
+- Skip reporting directories (meaning: make sure we know the type of a path)
+  - To do that, be smart and look whether a path refers to a file in a
+    subdirectory, then make sure that the subdirectory is marked as such
+  - Maybe over-report, and only mark known directories?
+  - Or not worry about this for now, it's okay to report directories, use
+    Trace2 to measure the impact of this
+
+- Exclude submodules somehow.
+
+- Handle when a file is removed and a directory added in its place: this needs
+  to be reported as "deleted".
+
+- Handle when a directory is renamed: only the directory name is reported, but
+  we need to report all paths.
+
+- Possibly prime the daemon's `paths` hashmap by reading the index and adding
+  all tracked files (and untracked files, if the untracked extension is
+  present)
```

Comment:
I guess your idea was to do a lazy `git status` (that writes an updated index) after a `git checkout` either via `git maintenance` or some other means. That sounds like a pretty cheap and robust solution.

Reply:
@dscho  By "prime the daemon's paths hashmap" did you mean the `strintern()` map that we're using or something else?   We only use that to cheaply de-dup paths that we send to the client.

Yes, the first few Git commands will be slow until one of them updates the index with a token that demonstrates that we have continuous observations of the file system with the currently active daemon.